src/common/split.h

```diff
@@ -0,0 +1,107 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#pragma once
+
+#include <string_view>
+
+namespace ceph {
+
+// a forward iterator over the parts of a split string
+class spliterator {
```

Comment:
how about moving these helpers into `str_list.h` , so we can start deprecating the `for_each_substr()` and friends?

Reply:
I'd suggest keeping it out of str_list (the name doesn't really fit) and eventually planning on removing that whole file.

Reply:
is there a reason it needs to be in str_list.h before we can deprecate those?

Reply:
not "needs to be", just thought it'd be easier to redirect new users to these `split*` utilies.